Dfe.Data.SearchPrototype/Web/Tests/APITests/ApiSearchResults.cs

```diff
@@ -56,22 +57,62 @@ public async Task GET_Search_Returns_EstablishmentData()
-        firstEstablishmentResult.Urn.Should().Be("123456");
-        firstEstablishmentResult.Name.Should().Be("Goose Academy");
-        firstEstablishmentResult.Address.Street.Should().Be("Goose Street");
-        firstEstablishmentResult.Address.Locality.Should().Be("Goose Locality");
-        firstEstablishmentResult.Address.Address3.Should().Be("Goose Address 3");
-        firstEstablishmentResult.Address.Town.Should().Be("Goose Town");
-        firstEstablishmentResult.Address.Postcode.Should().Be("GOO OSE");
-        firstEstablishmentResult.EstablishmentType.Should().Be("Academy");
-        firstEstablishmentResult.EstablishmentStatusName.Should().Be("Open");
-        firstEstablishmentResult.PhaseOfEducation.Should().Be("Secondary");
+        firstEstablishmentResult.Urn.Should().NotBeNull();
+        firstEstablishmentResult.Name.Should().NotBeNull();
+        firstEstablishmentResult.Address.Street.Should().NotBeNull();
+        firstEstablishmentResult.Address.Locality.Should().NotBeNull();
+        firstEstablishmentResult.Address.Address3.Should().NotBeNull();
+        firstEstablishmentResult.Address.Town.Should().NotBeNull();
+        firstEstablishmentResult.Address.Postcode.Should().NotBeNull();
+        firstEstablishmentResult.EstablishmentType.Should().NotBeNull();
+        firstEstablishmentResult.EstablishmentStatusName.Should().NotBeNull();
+        firstEstablishmentResult.PhaseOfEducation.Should().NotBeNull();
     }
 
-    [Fact]
-    public async Task GET_Search_NoMatch_Returns_NoEstablishmentData()
+    [Theory]
+    [InlineData("St")]
+    [InlineData("Jos")]
+    [InlineData("Cath")]
+    public async Task GET_Search_ByPartialName_ReturnsResults(string query)
+    {
+        var response = await _client.GetAsync($"{SEARCHKEYWORD_ENDPOINT}{query}");
+
+        var responseBody = await response.Content.ReadAsStringAsync();
+        var results = JsonConvert.DeserializeObject<EstablishmentResultsProperty>(responseBody)!;
+
+        results.EstablishmentResults!.Establishments.Should().HaveCountGreaterThan(1);
+
+        var establishmentResults = results.EstablishmentResults!.Establishments!;
+        foreach (var headings in establishmentResults!)
+        {
+            establishmentResults!.First().Name.Should().Contain(query);
+        }
+    }
+
+    [Theory]
+    [InlineData("Catholic")]
+    [InlineData("Junior")]
+    public async Task GET_Search_Returns_Facets(string query)
+    {
+        var response = await _client.GetAsync($"{SEARCHKEYWORD_ENDPOINT}{query}");
+
+        var responseBody = await response.Content.ReadAsStringAsync();
+        var results = JsonConvert.DeserializeObject<EstablishmentFacetsProperty>(responseBody)!;
+
+        var facets = results.EstablishmentFacetResults!.Facets!.Count();
+        facets.Should().Be(2);
+
+        var establishmentStatusName = results.EstablishmentFacetResults!.Facets!.SelectMany(t => t.Name!);
```

Comment:
this variable name `establishmentStatusName ` is misleading - maybe `facetName `would be better

Reply:
Changed to `establishmentStatus` as it's the heading displayed onscreen.